src/AutoMapper/TypeExtensions.cs

```diff
@@ -36,18 +36,35 @@ public static Type CreateType(this TypeBuilder type)
 
         public static IEnumerable<MemberInfo> GetAllMembers(this Type type)
         {
-            while (true)
+            if (type.IsInterface())
```

Comment:
I mean an extra test will always fix an issue, but that's not good code. Try to write general code that will work in all cases.